docs/development/pigs/pig-013.rst

```diff
@@ -0,0 +1,254 @@
+lot. Thanks to conda, we now have a way to ship up-to-date versions of Gammapy
+and all dependencies on all supported platforms, the number of installation
+issues reported by users has been very low in recent years.
+
+Given this situation, we feel that it is OK to drop support for old versions
+(most notably Python 2.7, see :ref:`pig-013`), and to change a few Gammapy
+dependencies from "optional" to "required". The difference is that a required
+dependency is in the ``install_requires`` list in ``setup.py``, which means that
+``pip install gammapy`` will install it, whereas optional dependencies are not
+listed there, and ``pip install gammapy`` will not automatically install them.
+Another distinction is that required dependencies are imported at the top of
+Gammapy modules, which means that import of some Gammapy sub-package such as
+e.g. ``import gammapy.detect`` will often fail should that dependency not be
+installed (which usually doesn't happen, unless users do uncommon things like
+e.g. ``pip uninstall scipy`` after ``pip install gammapy``, that would lead to a
+non-working Gammapy installation). Initially we only had Numpy and Astropy as
+required dependencies, by now we have added Scipy (in `GH 1863`_), PyYAML, click
+and regions as required dependencies.
+
+In Gammapy, we are careful and conservative concerning dependencies. E.g. we
+removed the use of `scikit-image`_ (in  `GH 1658`_) because really
+``scipy.ndimage`` contains everything we need, and also e.g. ``h5py`` and
+``pandas`` and ``ROOT`` and a few other optional dependencies that were only
+used in one or two functions in prototype code were removed (see e.g. `GH
+1586`_).
+
+In the following sections, we summarise what we have now (as of June 2019, in
+Gammapy v0.12), and changes we propose to be made immediately after the Gammapy
+v0.13 release in late July 2019, which will ship to users in Gammapy v0.14 in
+September 2019.
+
+As mentioned already in :ref:`pig-013`, there is always a benefit and a cost to
+supporting old versions. In a few sections below we highlight some of the
+benefits and costs for different dependencies specifically, but generally we
+feel that the benefits of the changes proposed here are significant (much
+reduced testing and maintenance effort, some new features become available),
+while the benefits of keeping support for older versions is very small, since
+most Gammapy users use the recommended conda environment, which already ships
+very modern and recent stack and is mostly unchanged. Especially the
+``astropy-regions`` and ``astropy-healpix`` packages are still under active
+development, and supporting old versions from 2018 or 2017 would be painful, and
+if we decide to require very recent versions of those packages (from summer
+2019), the motivation to support very old versions of other dependencies (most
+notably Astropy, but also e.g. Numpy or Scipy or Matplotlib) is very small.
+
+Required dependencies
+=====================
+
```

Comment:
Maybe but a short definition of "required dependency" here.

Reply:
Done  - but probably can be improved further. Give me a day, and then please read the PIG and comment again.